pkg/api/aim/runs.go

```diff
@@ -202,31 +210,43 @@ func GetRunMetrics(c *fiber.Ctx) error {
 	}
 
 	metrics := make(map[string]struct {
-		values []*float64
-		iters  []int
+		name    string
+		iters   []int
+		values  []*float64
+		context datatypes.JSON
 	}, len(metricKeys))
 	for _, m := range r.Metrics {
-		k := metrics[m.Key]
-
 		v := m.Value
 		pv := &v
 		if m.IsNan {
 			pv = nil
 		}
 
-		k.values = append(k.values, pv)
+		key := m.Key
+		if m.ContextID != nil {
+			key = fmt.Sprintf("%s%d", m.Key, *m.ContextID)
+		}
+		k := metrics[key]
+		k.name = m.Key
 		k.iters = append(k.iters, int(m.Iter))
-		metrics[m.Key] = k
+		k.values = append(k.values, pv)
+		if m.Context != nil {
+			k.context = m.Context.Json
+		} else {
```

Comment:
why can `k.context` have different types?

Reply:
because of endpoint nature. for normal(no stream) endpoints we can use it as is ->` m.Context.Json` but for the stream endpoint it just doesn't work and we have to re wrap it.